python/katana/local/import_data.py

```diff
@@ -50,12 +50,21 @@ def from_edge_list_matrix(edges: np.ndarray) -> Graph:
 @numba.njit()
 def _fill_indices(sort_order: np.ndarray, sources: np.ndarray, indices: np.ndarray):
     last_source = 0
-    for source_index, edge_index in enumerate(sort_order):
-        source = sources[source_index]
-        if source != last_source:
-            for j in range(last_source, source):
-                indices[j] = edge_index
-            last_source = source
+    # Code is duplicated because numba cannot perform type based cloning within a function and cannot pass ranges
+    # into a function.
+    if sort_order is None:
+        for edge_index, source in enumerate(sources):
+            if source != last_source:
```

Comment:
Is this calculating prefix sum? I don't completely understand how will this give prefix-sum for unsorted sources?

Reply:
For source[0, 10, 1] case, I am not sure how we will get indices[10] = 4? It is coming out to be 3 based on my rough calculations. I could be wrong.

Reply:
This code is only used in the presorted case. If sort_order is none the sources is already sorted.

Reply:
And to answer your prefix sum question: It's not really prefix sum per se. It's basically prefix sum and counting the out edges for a given node fused into one loop, now that I think of it. Maybe I should split it apart. What do you think?

Reply:
You need to compute edges per sources for prefix sum. I think it is fine to keep it in a single function. I was thinking  we can name it "_prefix_sum" but current name is fine as well.